From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: truong0235 <truong0235@users.noreply.github.com>
Date: Thu, 4 Sep 2026 12:00:00 +0700
Subject: [PATCH 1/4] fix: move SRGB to isWideColorMode true branch

The Poco C40 display only supports NATIVE and SRGB color modes. SRGB
was classified as non-wide, preventing populateColorModes() from
adding it to the mColorModes map. This caused getBestColorMode() to
always return NATIVE, bypassing the sRGB transfer function.

Moving SRGB to the isWideColorMode true branch allows it to be added
to the color modes map and selected by pickColorMode().
---
 services/surfaceflinger/SurfaceFlinger.cpp | 2 +1/-1
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 494aa2c..68e35e9 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -196,6 +196,7 @@ bool isWideColorMode(const ColorMode colorMode) {
         case ColorMode::DISPLAY_BT2020:
         case ColorMode::BT2100_PQ:
         case ColorMode::BT2100_HLG:
+        case ColorMode::SRGB:
             return true;
         case ColorMode::NATIVE:
         case ColorMode::STANDARD_BT601_625:
@@ -203,7 +204,6 @@ bool isWideColorMode(const ColorMode colorMode) {
         case ColorMode::STANDARD_BT601_525:
         case ColorMode::STANDARD_BT601_525_UNADJUSTED:
         case ColorMode::STANDARD_BT709:
-        case ColorMode::SRGB:
             return false;
     }
     return false;
-- 
2.55.0

